midend/hsIndexSimplify.h

```diff
@@ -0,0 +1,46 @@
+#ifndef _MIDEND_HSINDEXSIMPLIFY_H_
+#define _MIDEND_HSINDEXSIMPLIFY_H_
+
+#include "frontends/common/resolveReferences/referenceMap.h"
+#include "frontends/p4/typeChecking/typeChecker.h"
+#include "frontends/p4/typeMap.h"
+#include "ir/ir.h"
+
+namespace P4 {
+
+/// The main class for finding of non-concrete haader stack/arrya index.
+class HSIndexFinder : public Transform {
```

Comment:
A "finder" implies an Inspector.
This class must be doing more since it is a Transform. This suggests that it is not named properly.